File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum=nums[0]; int sum1=nums[0];
        for(int i=1;i<nums.length;i++){
            sum=Math.max(sum+nums[i],nums[i]);
            sum1=Math.max(sum,sum1);
        }
        return sum1;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(arr) { 
let sum=arr[0],sum1=arr[0]; 
  for(let i=1;i<arr.length;i++){
    sum=Math.max(sum+arr[i],arr[i]);
    sum1=Math.max(sum1,sum);
  }
  // console.log(sum);
  return sum1;
};